build/ctags.go

```diff
@@ -198,22 +199,24 @@ func ctagsAddSymbols(todo []*zoekt.Document, parser ctags.Parser, bin string) er
 	}
 
 	for k, tags := range fileTags {
-		symOffsets, err := tagsToSections(contents[k], tags)
+		symOffsets, symMetaData, err := tagsToSections(contents[k], tags)
 		if err != nil {
 			return fmt.Errorf("%s: %v", k, err)
 		}
 		todo[pathIndices[k]].Symbols = symOffsets
+		todo[pathIndices[k]].SymbolsMetaData = symMetaData
 		if len(tags) > 0 {
 			todo[pathIndices[k]].Language = strings.ToLower(tags[0].Language)
 		}
 	}
 	return nil
 }
 
-func tagsToSections(content []byte, tags []*ctags.Entry) ([]zoekt.DocumentSection, error) {
+func tagsToSections(content []byte, tags []*ctags.Entry) ([]zoekt.DocumentSection, []*zoekt.Symbol, error) {
 	nls := newLinesIndices(content)
 	nls = append(nls, uint32(len(content)))
 	var symOffsets []zoekt.DocumentSection
+	var symMetaData []*zoekt.Symbol
```

Comment:
maybe make these not a pointer, but a `[]zoekt.Symbol`. A bunch of pointers is more work for the garbage collector. If we can work out how to make zoekt.Symbol smaller (ie use enums instead of strings everywhere), that would be great :)